homeassistant/components/withings/__init__.py

```diff
@@ -36,50 +37,66 @@
 )
 
 
-async def async_setup(hass: HomeAssistantType, config: ConfigType):
+async def async_setup(hass: HomeAssistantType, config: ConfigType) -> bool:
     """Set up the Withings component."""
-    conf = config.get(DOMAIN)
+    conf = config.get(DOMAIN, {}).copy()
     if not conf:
         return True
 
-    hass.data[DOMAIN] = {const.CONFIG: conf}
-
     base_url = conf.get(const.BASE_URL, hass.config.api.base_url).rstrip("/")
+    hass.data[DOMAIN] = {const.CONFIG: {**conf, **{const.BASE_URL: base_url}}}
 
-    hass.http.register_view(config_flow.WithingsAuthCallbackView)
-
-    config_flow.register_flow_implementation(
+    config_flow.WithingsFlowHandler.async_register_implementation(
         hass,
-        conf[const.CLIENT_ID],
-        conf[const.CLIENT_SECRET],
-        base_url,
-        conf[const.PROFILES],
-    )
-
-    hass.async_create_task(
-        hass.config_entries.flow.async_init(
-            DOMAIN, context={"source": SOURCE_IMPORT}, data={}
-        )
+        config_entry_oauth2_flow.LocalOAuth2Implementation(
+            hass,
+            const.DOMAIN,
+            conf[const.CLIENT_ID],
+            conf[const.CLIENT_SECRET],
+            "%s/%s" % (WithingsAuth.URL, "oauth2_user/authorize2"),
```

Comment:
Use string formatting

```suggestion
            f"{WithingsAuth.URL}/oauth2_user/authorize2",
```